src/features/create-dao/steps/DetailsForm.tsx

```diff
@@ -0,0 +1,117 @@
+import { FC, useContext } from 'react';
+
+import * as Yup from 'yup';
+import { Form, Formik } from 'formik';
+
+import { MediaInput, MediaType } from '@zero-tech/zui/components/MediaInput';
+import { Wizard } from '@zero-tech/zui/components';
+import { WrappedInput } from '../../ui/WrappedInput/WrappedInput';
+import { CreateDAOFormContext } from '../';
+
+import styles from './DetailsForm.module.scss';
+import classNames from 'classnames/bind';
+
+const validationSchema = Yup.object().shape({
+	name: Yup.string().required('The name field is required.'),
+	znaAddress: Yup.string().required('The zNA address field is required.'),
+	description: Yup.string().required('The description field is required.'),
+});
+
+export interface DetailsFormProps {
+	onClose: () => void;
+}
+
+export const DetailsForm: FC<DetailsFormProps> = ({ onClose }) => {
+	const { details, onDetailsSubmit } = useContext(CreateDAOFormContext);
+
+	const handleMediaInputChange = (
+		mediaType: MediaType,
+		previewUrl: string,
+		image: Buffer,
+		setFieldValue: (
+			field: string,
+			value: any,
+			shouldValidate?: boolean,
+		) => void,
+	): void => {
+		setFieldValue('mediaType', mediaType);
+		setFieldValue('previewUrl', previewUrl);
+		setFieldValue('avatar', image);
+	};
+
+	return (
+		<Formik
+			initialValues={details}
+			onSubmit={(values) => onDetailsSubmit(values)}
+			validationSchema={validationSchema}
+		>
+			{({ values, errors, touched, setFieldValue, submitForm }) => (
+				<Form>
+					<div className={styles.Row}>
+						<div className={styles.Column}>
+							<MediaInput
+								className={styles.MediaInput}
+								title="Upload DAO avatar..."
+								subtitle="(Optional)"
+								mediaType={values.mediaType}
+								previewUrl={values.previewUrl}
+								hasError={false}
```

Comment:
something I've just picked up - should the prop be a conditional rather than hardcode true/false?

Reply:
Good spotting. Have added state to handle the avatar error!